docs/specifications/tests/Zone-TP/zone11.md

```diff
@@ -0,0 +1,166 @@
+Z11_MNAME_NO_RESPONSE         | NOTICE  | ns                   | MNAME name server points to a non responsive name server ("{ns}").
+Z11_MNAME_NOT_IN_GLUE         | NOTICE  |                      | MNAME name server could not be found in the parent zone name servers.
+Z11_MNAME_NOT_MASTER          | WARNING | ns_ip_list           | MNAME name server has a mis-matching SERIAL. Returned from name servers "{ns_ip_list}".
+Z11_MNAME_IS_MASTER           | INFO    | ns_ip_list           | MNAME name server is master. Returned from name servers "{ns_ip_list}".
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+## Test procedure
+
+In this section and unless otherwise specified below, the terms "[DNS Query]"
+follow the specification for DNS queries as specified in [DNS Query and Response Defaults].
+The handling of the DNS responses on the DNS queries follow, unless otherwise specified below,
+what is specified for [DNS Response] in the same specification.
+
+1. Create a [DNS Query] with query type SOA and query name *Child Zone* ("SOA Query")
+
+2. Obtain the set of name server IP addresses using [Method5]
+   ("Name Server IP").
+
+3. Create the following empty sets:
+   1. Name server IP address ("No MNAME Record")
+   2. Name server IP address ("No Serial Record")
+   3. Name server IP address ("MNAME Is Zone Name")
+   4. Name server IP address ("MNAME Not Master")
+   5. Name server IP address ("MNAME Is Master")
+
+4. For each name server IP in *Name Server IP* do:
+   1. Send *SOA Query* to the name server and collect the response.
+   2. Go to next name server IP if at least one of the following criteria is met:
+      1. There is no DNS response.
+      2. [RCODE Name] of the response is not "NoError".
+      3. The AA flag is not set in the response.
+      4. There is no SOA record with owner name matching the query.
+   3. From the DNS response, store the MNAME and its associated SERIAL.
+      1. Go to next name server IP if at least one of the following criteria is met:
+         1. If there is no MNAME record, then add name server IP to the *No MNAME Record* set. 
+         2. If there is no SERIAL record, then add name server IP to the *No SERIAL Record* set. 
+         3. If the name in the MNAME record is the same as *Child Zone*, then add name server IP to the *MNAME Is Zone Name* set.
+   4. Send *SOA Query* to the name server in the MNAME ("MNAME Nameserver").
+      1. If the SOA response has [RCODE Name] "NoError" then:
+         1. If the AA flag is not set, then output *Z11_MNAME_NOT_AUTHORITATIVE* with name server name and IP.
+      2. Else, output *Z11_MNAME_NO_RESPONSE* with name server name and IP and go to next name server IP.

```

Comment:
Wouldn't it be more efficient to collect all MNAME found in the loop in 4 starting on line 76, and then send a query to each of them? In most cases there will be only one. (Consistency is covered by Consistency06).

Reply:
Updated